pre-docs/writing_tasks.md

```diff
@@ -188,23 +188,36 @@ A task can also have multiple implementations, with metadata that explains when
 
 ```
 - tasks
-  - sql.sh
-  - sql.ps1
+  - sql_linux.sh
+  - sql_linux.json
+  - sql_windows.ps1
+  - sql_windows.json
   - sql.json
 ```
 
-This task has two executables \(sql.sh and sql.ps1\) with a metadata file. The metadata file contains an implementations section:
+This task has two executables \(sql_linux.sh and sql_windows.ps1\) with a metadata file. The executables have distinct names, so that they can be uploaded to the forge. Each implementation can also optionally have it's own metadata, in addition to the task metadata. This provides information about the implementations, which can be seen in the Puppet Enterprise Tasks section and help others determine which implementation to use.
```

Comment:
- I think it would be best to explain the underlying reason "so that they can be used with older task runners such as Puppet Enterprise before 2019".  
- Each implementation must have it's own metadata to be uploaded to forge. now that private has landed soemthing like:  "Task authors should provide metadata for each implementation either documenting how to use it directly or marking it private to hide it from UI lists.